### PR TITLE
Add texture handling.

### PR DESCRIPTION
**Pull request by LumpBloom7**
Adds basic texture functionality.

* `Texture` - Basic class holding an `SDL_Texture *` and a `render()` function
* `TextureStore` - Class to load and store `Texture`s. It assignes shared_ptr<Texture> and prevents creating a new Texture from the same file.

### CODE REVIEW

File: BloomFramework/src/TextureStore.cpp [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+			}
+			//Create texture from surface pixels
+			SDL_Texture * texture = SDL_CreateTextureFromSurface(*m_renderer, loadedSurface);
+			if (texture == nullptr)
+			{
+				throw Exception("[SDL_Texture] " + std::string(SDL_GetError()));
+			}
+			else {
+				TexturePtr ptr = std::make_shared<Texture>(texture, m_renderer);
+				m_store.emplace(filePath, ptr);
```
Comment (siew24) [inline BloomFramework/src/TextureStore.cpp:31]: Be sure to have enough storage in m_store

File: BloomFramework/src/TextureStore.cpp [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+				SDL_FreeSurface(loadedSurface);
+				return ptr;
+			}
+		}
+	}
+
+	TexturePtr TextureStore::getTexture(const std::string & filePath) {
+		auto texIterator = m_store.find(filePath);
+		if (texIterator != m_store.end())
```
Comment (siew24) [inline BloomFramework/src/TextureStore.cpp:40]: texIterator won't be neccessary

Comment (siew24) [inline BloomFramework/src/TextureStore.cpp:40]: texIterator won't be necessary.

**Review by siew24 — APPROVED** [review]
For now i don't see any possible improvements :+1:

File: BloomFramework/src/TextureStore.cpp (hunk 1 above)
Comment (LumpBloom7) [reply BloomFramework/src/TextureStore.cpp:31]: That is dependent on system RAM, but normally should exceed limits unless someone bypasses the duplicate texture check or loads an absurd amount of textures.

File: BloomFramework/src/TextureStore.cpp (hunk 2 above)
Comment (LumpBloom7) [reply BloomFramework/src/TextureStore.cpp:40]: It let's us return it immediately rather than needing to find the TexturePtr to return.

**Pull request merged**